numpy/testing/_private/utils.py

```diff
@@ -404,6 +389,31 @@ def assert_equal(actual, desired, err_msg='', verbose=True):
     except (TypeError, ValueError, NotImplementedError):
         pass
 
+    # Inf/nan/negative zero handling
+    try:
+        isdesnan = gisnan(desired)
+        isactnan = gisnan(actual)
+        if isdesnan and isactnan:
+            return  # both nan, so equal
+
+        # handle signed zero specially for floats
+        array_actual = array(actual)
+        array_desired = array(desired)
+        if (array_actual.dtype.char in 'Mm' or
+                array_desired.dtype.char in 'Mm'):
```

Comment:
You only want `M` here, not `m`

Reply:
It doesn't really matter, the test just below this is only for things that will have a signbit on a 0 value, so timedelta64 might as well never go to the next clause

Reply:
Well at that point you may as well add integers here too - at any rate, the comment is saying "this won't work", not "this is a shortcut to save time".

Reply:
ok, fixing

Reply:
actually, `np.timedelta64(7, 's') == 0` emits a DeprecationWarning, so we need both 'Mm', and the comment is wrong

Reply:
Seems weird that `td[s] == 0` is deprecated yet `td[s] + 0` is not - perhaps worth opening a new issue about.